packages/block-library/src/common.scss

```diff
@@ -161,3 +161,22 @@ html :where(img[class*="wp-image-"]) {
 figure {
 	margin: 0 0 1em 0;
 }
+
+// Set the admin bar offset for sticky positioned blocks to the height of the admin bar.
+// This allows sticky positioned blocks to be positioned correctly when the admin bar is visible.
+html :where(.is-position-sticky) {
+	/* stylelint-disable length-zero-no-unit */
+	--wp-admin--admin-bar--position-offset: var(--wp-admin--admin-bar--height, 0px); // 0px is set explicitly so that it can be used in a calc value.
+	/* stylelint-enable length-zero-no-unit */
+}
+
+// To support sticky positioning, reset the admin bar offset to 0px on mobile devices,
+// within the scope of the position classname. This is required because the admin bar
+// is not fixed to the top on mobile devices, but is fixed on viewports larger than 600px.
+@media screen and (max-width: 600px) {
+	.is-position-sticky {
```

Comment:
Nit: would be nice to either use the `:where()` or not both inside the media query and out.

Reply:
Oh, good point, I'll add it here, too

Reply:
Updated the selector in https://github.com/WordPress/gutenberg/pull/47665/commits/52bd2b342d63d323683f8f34a0c42c0c37875fcb